web/src/pages/Dashboard.tsx

```diff
@@ -1,121 +1,168 @@
-import React, { useRef, useState } from "react"
+import { useState, useEffect } from "react"
 import { useTranslation } from "react-i18next"
 
 import { useUser } from "../contexts/UserContext"
-import PieCharts from "../components/PieCharts"
+import type { User} from "../contexts/UserContext"
+import { ApiClient } from "@transcenders/api-client"
+// import PieCharts from "../components/PieCharts"
 import AreaCharts from "../components/AreaCharts"
 import MatchHistory from "../components/MatchHistory"
 
+import avatarCat1 from "/images/avatarCat1.avif"
+import avatarCat2 from "/images/avatarCat2.avif"
+import avatarCat3 from "/images/avatarCat3.avif"
+import avatarCat4 from "/images/avatarCat4.avif"
+import avatarCat5 from "/images/avatarCat5.avif"
+import avatarCat6 from "/images/avatarCat6.avif"
+
+const avatars = [avatarCat1, avatarCat2, avatarCat3, avatarCat4, avatarCat5, avatarCat6]
+
 const Dashboard = () => {
-  const { t } = useTranslation()
-  const { user } = useUser()
-
-  const [avatar, setAvatar] = useState<File | null>(null)
-  const [previewUrl, setPreviewUrl] = useState<string | null>(null)
-  const fileInputRef = useRef<HTMLInputElement>(null)
-
-  const handleFileChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-    const file = e.target.files?.[0];
-    if (file) {
-      setAvatar(file);
-      setPreviewUrl(URL.createObjectURL(file));
-    }
-  };
-
-  const handleUploadClick = () => {
-    fileInputRef.current?.click()
-  }
-
-  const wins = 42
-  const losses = 13
-
-  return (
-		<div className="w-full h-full flex gap-6">
-			<div className="profile-box basis-1/5 justify-between p-6 py-16">
+	const { t } = useTranslation()
+	const { user } = useUser()
+
+	const avatarSizes = [
+		"max-w-[65%]",
+		"max-w-[85%]",
+		"max-w-[98%]",
+		"max-w-[80%]",
+		"max-w-[80%]",
+		"max-w-[70%]",
+	]
+
+	const friends = [
+		{ id: 1, name: "Mittens", online: true },
+		{ id: 2, name: "Shadow", online: false },
+		{ id: 3, name: "Whiskers", online: true },
+		{ id: 4, name: "Mittens", online: true },
+		{ id: 5, name: "Shadow", online: false },
+		{ id: 6, name: "Whiskers", online: true },
+		{ id: 7, name: "Mittens", online: true },
+		{ id: 8, name: "Shadow", online: false },
+		{ id: 9, name: "Whiskers", online: true },
+		{ id: 10, name: "Mittens", online: true },
+		{ id: 11, name: "Shadow", online: false },
+		{ id: 12, name: "Whiskers", online: true },
+		{ id: 13, name: "Mittens", online: true },
+		{ id: 14, name: "Shadow", online: false },
+		{ id: 15, name: "Whiskers", online: true },
+	]
+
+	const [viewedUser, setViewedUser] = useState<User | null>(user)
+	const [searchId, setSearchId] = useState("")
+	const [error, setError] = useState("")
+
+	const handleSearch = async () => {
+		setError("")
+		const id = parseInt(searchId)
+
+		try {
+			const response = await ApiClient.user.getUserById(id)
```

Comment:
To search by name instead, something like this should work

```suggestion
      const response = await ApiClient.user.getUserExact({ username: searchId });
```
there is also the `ApiClient.user.getAllUsers({ search:searchId })` ( I should rename it to `getUsers`) that will find things that consist of the serach term anywhere in the username, email or displayname, for example when my search term is 'a' it will give all users that have letter a anywhere in the username, email or displayname, can be used for sort of a live search thing I guess, until it pinpoints an exact user .